joblib/test/test_format_stack.py

```diff
@@ -20,3 +21,23 @@ def __repr__(self):
 
 def test_safe_repr():
     safe_repr(Vicious())
+
+def test_format_records():
+    try:
+        raise ValueError
+    except ValueError:
+        etb = sys.exc_info()[2]
+        records = _fixed_getframes(etb)
+
+        # Modify filenames in traceback records from .py to .pyc
+        pyc_records = []
+        for record in records:
+            if record[1].endswith('.py'):
+                record = list(record)
+                record[1] += 'c'
+                record = tuple(record)
+                pyc_records.append(record)
```

Comment:
This is the kind of place list comprehensions shine IMHO:

```python
pyc_records = [(_1, filename + 'c' if filename.endswith('.py') else filename,
                _2, _3, _4, _5)
               for _1, filename, _2, _3, _4, _5 in records]
```

In this particular case maybe using a helper function would make it clearer.